filebeat/module/nginx/access/ingest/default.json

```diff
@@ -43,17 +51,24 @@
         {
             "script": {
                 "lang": "painless",
-                "source": "boolean isPrivate(def dot, def ip) { try { StringTokenizer tok = new StringTokenizer(ip, dot); int firstByte = Integer.parseInt(tok.nextToken());       int secondByte = Integer.parseInt(tok.nextToken());       if (firstByte == 10) {         return true;       }       if (firstByte == 192 && secondByte == 168) {         return true;       }       if (firstByte == 172 && secondByte >= 16 && secondByte <= 31) {         return true;       }       if (firstByte == 127) {         return true;       }       return false;     } catch (Exception e) {       return false;     }   }   def found = false;   for (def item : ctx.nginx.access.remote_ip_list) {     if (!isPrivate(params.dot, item)) {       ctx.source.ip = item;       found = true;       break;     }   }   if (!found) {     ctx.source.ip = ctx.nginx.access.remote_ip_list[0];   }",
+                "source": "boolean isPrivate(def dot, def ip) { try { StringTokenizer tok = new StringTokenizer(ip, dot); int firstByte = Integer.parseInt(tok.nextToken());       int secondByte = Integer.parseInt(tok.nextToken());       if (firstByte == 10) {         return true;       }       if (firstByte == 192 && secondByte == 168) {         return true;       }       if (firstByte == 172 && secondByte >= 16 && secondByte <= 31) {         return true;       }       if (firstByte == 127) {         return true;       }       return false;     } catch (Exception e) {       return false;     }   }   try {    ctx.source.ip = null;    if (ctx.nginx.access.remote_ip_list == null) { return; }    def found = false;    for (def item : ctx.nginx.access.remote_ip_list) {        if (!isPrivate(params.dot, item)) {            ctx.source.ip = item;            found = true;            break;        }    }    if (!found) {     ctx.source.ip = ctx.nginx.access.remote_ip_list[0];   }} catch (Exception e) { ctx.source.ip = null; }",
```

Comment:
It is a bit difficult to see changes here :grimacing:, but, why setting `source.ip` to null if later it is removed in that case?

Reply:
Good question and sorry! I remember to think about "I'm gonna copy-paste pretty code on PR" and I finally forgot it

```java
boolean isPrivate(def dot, def ip) {
    try {
        StringTokenizer tok = new StringTokenizer(ip, dot);
        int firstByte = Integer.parseInt(tok.nextToken());
        int secondByte = Integer.parseInt(tok.nextToken());
        if (firstByte == 10) {         return true;       }
        if (firstByte == 192 && secondByte == 168) {         return true;       }
        if (firstByte == 172 && secondByte >= 16 && secondByte <= 31) {         return true;       }
        if (firstByte == 127) {         return true;       }
        return false;
    } catch (Exception e) {
        return false;
    }
}

try {
    ctx.source.ip = null;
    if (ctx.nginx.access.remote_ip_list == null) { return; }
    def found = false;
    for (def item : ctx.nginx.access.remote_ip_list) {
        if (!isPrivate(params.dot, item)) {
            ctx.source.ip = item;
            found = true;
            break;
        }
    }
    
    if (!found) {
        ctx.source.ip = ctx.nginx.access.remote_ip_list[0];
    }
} catch (Exception e) {
    ctx.source.ip = null;
}
```

The thing is that "convert" step in the pipeline was failing by not recognizing "" as a valid string with the error:

```json
{"type":"mapper_parsing_exception","reason":"failed to parse field [source.ip] of type [ip] in document with id '6SrYOWkBxc4oeub4XID4'","caused_by":{"type":"illegal_argument_exception","reason":"'' is not an IP string literal."}
```
But an `ignore_missing` works if the field  doesn't exists or it's null. It doesn't work if field is just empty and `ignore_failure` seemed to just omit the entire line (not the processing step).

I have to admit that it was terribly difficult

Reply:
Now it seems that we can use yaml here :relieved: #11209